drainer/checkpoint/file.go

```diff
@@ -92,6 +93,7 @@ func (sp *FileCheckPoint) Save(ts, secondaryTS int64, consistent bool) error {
 
 	sp.CommitTS = ts
 	sp.ConsistentSaved = consistent
+	sp.Version = version
```

Comment:
Should we prevent the version being reduced here?

Reply:
A recently test case shows that drainer writes the old schema version again after manually adjusting the schema version to the correct value.

Reply:
why?

Reply:
The test case with amend transaction enabled.

|dml|ddl|schema version|
|-|-|-|
|begin||1|
|insert...|||
||ddls|from 1 to 10|
|commit||10|

The ddls are executed before transaction committed and push the schema version to 10 in checkpoint. When the drainer is executing the dml transaction, the schema version of it is still 1, and will checkpoint with schema version 1 into checkpoint.